File: BaekJoon/src/step5/Problem_4673.java
package step5;

public class Problem_4673 {
	int self_num(int a) {
		int sum = a;
		
		while(a != 0) {
			sum += a % 10;
			a /= 10;
		}
		
		return sum;
	}
	
	public static void main(String[] args) {
		int a = 1;
		boolean[] temp = new boolean[10001];
		Problem_4673 pro = new Problem_4673();
		
		while(true) {
			
			int s = pro.self_num(a);
			if(a>10001) {
				break;
			}
			if(s <=10000) {
				temp[s] = true;
			}
			
			a++;
		}
		
		for(int i=1; i<10001;i++) {
			if(!temp[i]) {
				System.out.println(i);
			}
		}
	}
}

File: BaekJoon/src/step5/Problem_4673.py
temp = [0] * 10001
a = 1

def self_num(a):
    sum = a
    while a != 0:
        sum += a%10
        a //= 10

    return sum

while True:
    
    s = self_num(a)
    
    if a>10001:
        break

    a += 1
    
    if s <= 10000:
        temp[s] = 1


for i in range(1,10001):
    if temp[i] != 1:
        print(i)
